scripts/test-uis.py

```diff
@@ -0,0 +1,302 @@
+        else:
+            logging.error(f'Found unexpected console log {log.type}. Text: "{log.text}", url: {url}')
+
+            has_err = True
+
+    if has_err:
+        raise RuntimeError('Console has unexpected warnings and/or errors. Check previous logs')
+
+    logging.info('Console logs are ok')
+
+
+def _click_button(page, button):
+    logging.info(f'Clicking button "{button}"')
+    page.get_by_role('button', name=button).click()
+
+
+def _set_text(page, text, value):
+    logging.info(f'Setting text to "{value}"')
+    page.get_by_role('textbox', name=text).fill(value)
+
+
+def _test_frontend_ui_handler(page):
+    # Set the name & Click the button
+    name = 'my name'
+    logging.info(f'Setting name "{name}"')
+    page.get_by_label('Enter your name:').fill(name)
+    _click_button(page, 'Click Me!')
+
+    # Check if `#greeting` is populated correctly
+    greeting_id = '#greeting'
+    timeout_ms = 60000
+    greeting_obj = page.wait_for_selector(greeting_id, timeout=timeout_ms)
+    if greeting_obj:
+        actual_value = greeting_obj.inner_text()
+        expected_value = f'Hello, {name}!'
+        if actual_value == expected_value:
+            logging.info(f'"{actual_value}" found in "{greeting_id}"')
+        else:
+            raise RuntimeError(f'Expected greeting message is "{expected_value}", but found "{actual_value}"')
+    else:
+        raise RuntimeError(f'Cannot find {greeting_id} selector')
+
+
+def _test_candid_ui_handler(page):
+    # Set the text & Click the "Query" button
+    text = 'hello, world'
+    _set_text(page, 'text', text)
+    _click_button(page, 'Query')
+
+    # Check if `#output-list` is populated correctly (after the first click)
+    output_list_id = '#output-list'
+    timeout_ms = 60000
+    _ = page.wait_for_selector(output_list_id, timeout=timeout_ms)
+
+    # Reset the text & Click the "Random" button
+    _set_text(page, 'text', '')
+    _click_button(page, 'Random')
+    # ~
+
+    # Check if `#output-list` is populated correctly  (after the second click)
+    output_list_obj = page.wait_for_selector(output_list_id, timeout=timeout_ms)
+    if output_list_obj:
+        output_list_lines = output_list_obj.inner_text().split('\n')
+        actual_num_lines, expected_num_lines = len(output_list_lines), 4
+        if actual_num_lines != expected_num_lines:
+            err = [f'Expected {expected_num_lines} lines of text but found {actual_num_lines}']
+            err.append('Lines:')
+            err.extend(output_list_lines)
+            raise RuntimeError('\n'.join(err))
+
+        # Extract random text from third line
+        random_text = re.search(r'"([^"]*)"', output_list_lines[2])
+        if not random_text:
+            raise RuntimeError(f'Cannot extract the random text from the third line: {output_list_lines[2]}')
+        random_text = random_text.group(1)
+
+        for i, text_str in enumerate([text, random_text]):
+            l1, l2 = (i * 2), (i * 2 + 1)
```

Comment:
`l` and `1` are too close to be used like this in a variable name

```suggestion
            line1, line2 = (i * 2), (i * 2 + 1)
```